com.arc.embeddedcdt/src/com/arc/embeddedcdt/gui/ARCTerminalTab.java

```diff
@@ -265,19 +269,13 @@ protected String getAttributeValueFromCombo(Combo combo) {
     public static List COMserialport() {
         List<String> list = new ArrayList<String>();
         try {
-            Enumeration portIdEnum = CommPortIdentifier.getPortIdentifiers();
-            while (portIdEnum.hasMoreElements()) {
-                CommPortIdentifier identifier = (CommPortIdentifier) portIdEnum.nextElement();
-                String strName = identifier.getName();
-                int nPortType = identifier.getPortType();
-
-                if (nPortType == CommPortIdentifier.PORT_SERIAL)
-                    list.add(strName);
+            String[] ports = SerialPort.list();
+            for(String strName : ports) {
+                list.add(strName);
             }
-
-        } catch (IllegalArgumentException e1) {
-            // TODO Auto-generated catch block
-            e1.printStackTrace();
+        } catch (IOException e) {
+        	//TODO createMarker and other stuff
+        	StatusManager.getManager().handle(new Status(IStatus.ERROR, LaunchPlugin.PLUGIN_ID, "Debugger plugin has thrown IOException", e), StatusManager.SHOW);
```

Comment:
ERR_LINE_TOO_LONG

Reply:
Use more clear error message, something like: "An error happened when retrieving list of available serial ports."